Refactor structure and fix broken links in documentation

saumyashahi opened the pull request:
## Summary
Fix issues in the documentation by resolving broken links caused by incorrect Markdown references and reorganizing misplaced text in the toolbar and context menu sections.
- Fixes #4235, #2403

## Key Changes
- Correct Markdown syntax for link references to make them valid.
- Move the context menu text from the toolbar section to its appropriate location in the context menu section.
- Adjust structure for clarity and accuracy to ensure sections align with their intended topics.

## Testing
View the rendered documentation to confirm that the changes appear correctly in the desired format.

saumyashahi commented:
@walterbender, I have addressed the issues and made the necessary changes. Please review the updates and let me know if there are any additional adjustments or improvements needed.

walterbender commented:
This is an improvement. But I think that have the block context described in two different places is a bit confusing and redundant. Maybe consolidate them?

saumyashahi commented:
I have made the changes and removed redundancy. I also added this labelled image to give a better understanding of background context menu:

<img src="https://github.com/user-attachments/assets/76041c97-3f23-4183-a14e-065870af3da3" alt="Context Menu for Workspace" width="60%">

walterbender commented:
Much much nicer. Thx.

saumyashahi commented:
Thank you, @walterbender sir! I'm glad to have helped.
